lib/discourse_code_review/state/github_repo_categories.rb

```diff
@@ -14,6 +15,27 @@ def ensure_category(repo_name:)
                 .where(name: GITHUB_REPO_NAME, value: repo_name)
           ).first
 
+        if category.present? && category.custom_fields[GITHUB_REPO_ID].blank? && repo_id.present?
+          category.custom_fields[GITHUB_REPO_ID] = repo_id
+          category.save_custom_fields
+        end
+
+        if category.blank? && repo_id.present?
+          category =
+            Category.where(
+              id:
+                CategoryCustomField
+                  .select(:category_id)
```

Comment:
`pluck` would be better here since you don't need the entire object.

Reply:
Replacing `select` with `pluck` breaks various tests as they are using the category object to check for various attributes.

Reply:
@eviltrout, `select` doesn't actually fetch anything here. When it's used in a subquery, it's turned into SQL.

Reply:
That's very cool